tests/parametric/test_otel_logs.py

```diff
@@ -0,0 +1,631 @@
+        """Custom headers from OTEL_EXPORTER_OTLP_HEADERS appear in requests."""
+        with test_library as library:
+            library.write_log("test_custom_http_headers_included_in_otlp_export", LogLevel.INFO, "test_logger")
+
+        log_payloads = test_agent.wait_for_num_log_payloads(1)
+        assert (
+            find_log_record(log_payloads, "test_logger", "test_custom_http_headers_included_in_otlp_export") is not None
+        )
+
+        requests = test_agent.requests()
+        logs_request = [r for r in requests if r["url"].endswith("/v1/logs")]
+        assert logs_request, f"Expected logs request, got {requests}"
+        assert logs_request[0]["headers"].get("api-key") == "key", f"Expected api-key, got {logs_request[0]['headers']}"
+        assert (
+            logs_request[0]["headers"].get("other-config-value") == "value"
+        ), f"Expected other-config-value, got {logs_request[0]['headers']}"
+
+    @pytest.mark.parametrize(
+        "library_env",
+        [
+            {
+                "DD_LOGS_OTEL_ENABLED": "true",
+                "DD_TRACE_DEBUG": None,
+                "OTEL_EXPORTER_OTLP_LOGS_HEADERS": "api-key=key,other-config-value=value",
+                "OTEL_EXPORTER_OTLP_PROTOCOL": "http/protobuf",
+            },
+        ],
+    )
+    def test_custom_logs_http_headers_included_in_otlp_export(self, test_agent, test_library, library_env):
+        """Custom headers from OTEL_EXPORTER_OTLP_LOGS_HEADERS appear in requests."""
+        with test_library as library:
+            library.write_log("test_custom_logs_http_headers_included_in_otlp_export", LogLevel.INFO, "test_logger")
+
+        log_payloads = test_agent.wait_for_num_log_payloads(1)
+        assert (
+            find_log_record(log_payloads, "test_logger", "test_custom_logs_http_headers_included_in_otlp_export")
+            is not None
+        )
+
+        requests = test_agent.requests()
+        logs_request = [r for r in requests if r["url"].endswith("/v1/logs")]
+        assert logs_request, f"Expected logs request, got {requests}"
+        assert logs_request[0]["headers"].get("api-key") == "key", f"Expected api-key, got {logs_request[0]['headers']}"
+        assert (
+            logs_request[0]["headers"].get("other-config-value") == "value"
+        ), f"Expected other-config-value, got {logs_request[0]['headers']}"
+
+
+@features.otel_logs_enabled
+@scenarios.parametric
+class Test_FR09_Log_Injection:
+    """FR09: Log Injection Tests"""
+
+    @pytest.mark.parametrize(
+        "library_env",
+        [
+            {
+                "DD_LOGS_OTEL_ENABLED": "true",
+                "DD_SERVICE": "testservice",
+                "DD_ENV": "testenv",
+                "DD_VERSION": "1.0.0",
+                "DD_TRACE_DEBUG": None,
+            },
+        ],
+    )
+    def test_log_injection_disabled_when_otel_enabled(self, test_agent, test_library, library_env):
```

Comment:
nit: Asserting that a feature (Log Injection) is disabled feels weird to me. Can we make it affirmative by renaming it to testing that the subfeature "Trace Log Correlation" works as expected?